### DISCUSSION TITLE
Potential issue with player data not getting wiped after logging out

### DISCUSSION THREAD

**Discussion by mathiasHillmann** [runelite/runelite#16964] [Q&A] [answered]
Currently before the first login the account hash is set to -1 and after the first login it is set to the current player's hash. But after logging out and in the login screen the account hash outputted by `client.getAccountHash()` is still the last logged in player. Is this correct behavior or should the account hash be wiped on logout?

`client.getLocalPlayer()` has the same behavior, being null before the first login but set to the last logged in player until another player logs in.

**Comment by Adam-** [comment, marked as answer]
The Jagex client just does not null it. You are supposed to check the game state and consider them invalid if not logged in. We've never bothered patching it because it did not seem necessary.
